### DISCUSSION TITLE
Dev server start

### DISCUSSION THREAD

**Discussion by fac3m4n** [rubiin/ultimate-nest#488] [Q&A] [answered]
After starting server and trying to access /doc for swagger, I am having this error. Can you help understaing it?
![Screenshot 2024-08-27 at 13 56 48](https://github.com/user-attachments/assets/f96e5e54-ddd9-4b57-a745-a8ad6f066b33)

**Comment by fac3m4n** [comment]
API itself works:
<img width="936" alt="image" src="https://github.com/user-attachments/assets/fc3a633a-7764-48f6-8c80-06d11524b23c">

**Reply by rubiin** [reply]
Thats strange. Are there any files you included. I could not replicate it
If this still persists can you push your code to github and share me the link

**Reply by fac3m4n** [reply]
Here you go:

[https://github.com/fac3m4n/ultimate-nest](https://github.com/fac3m4n/ultimate-nest)

**Reply by fac3m4n** [reply]
Think its working
<img width="1500" alt="image" src="https://github.com/user-attachments/assets/33972d89-e080-4e63-a0da-fe67b7696fdd">

**Reply by mndev222** [reply]
can you share your .env and are you running postgres on docker?  Found your repo, beacause the one you linked doesn't work. 
https://github.com/fac3m4n/ultimate-nest/commits/master/ I see your commits, and can't find a code you changed to make it working.

**Reply by fac3m4n** [reply]
Yea I am using postgres on docker. Swagger docs are there but other routes are not showing. Also updated the link. Here is env.
`APP_PORT=3001
APP_PREFIX=v1
APP_NAME=teez
NODE_ENV=dev
API_URL=http://localhost
CLIENT_URL=http://google.com
SWAGGER_USER=user
ALLOWED_HOSTS=*
SWAGGER_PASSWORD=swagger

	# database
DB_HOST=localhost
DB_PORT=5432
DB_USERNAME=root
DB_PASSWORD=root
DB_DATABASE=ultimate`

I have sentry and cloudinary keys, which is ok

**Reply by mndev222** [reply]
https://github.com/mndev222/ultimate-nestjs-test1/commit/64fbbb88b13027faa1be5e1e8ff90371e68d3508 Here's my commit what I changed. If I would like to run a dev server on docker I run:
`ENV=dev sh ./scripts/deploy.sh`
I got an error:
```bash
Already up to date.
Deployment Command: docker compose up nestify -d
time="2024-08-27T15:41:40+02:00" level=warning msg="The \"DB_USERNAME\" variable is not set. Defaulting to a blank string."
time="2024-08-27T15:41:40+02:00" level=warning msg="The \"DB_PASSWORD\" variable is not set. Defaulting to a blank string."
time="2024-08-27T15:41:40+02:00" level=warning msg="The \"DB_DATABASE\" variable is not set. Defaulting to a blank string."
time="2024-08-27T15:41:40+02:00" level=warning msg="The \"API_URL\" variable is not set. Defaulting to a blank string."
time="2024-08-27T15:41:40+02:00" level=warning msg="The \"RABBITMQ_DEFAULT_USER\" variable is not set. Defaulting to a blank string."
time="2024-08-27T15:41:40+02:00" level=warning msg="The \"RABBITMQ_DEFAULT_PASS\" variable is not set. Defaulting to a blank string."
service "redis" refers to undefined volume config/redis.conf: invalid compose project
"docker rmi" requires at least 1 argument.
See 'docker rmi --help'.

Usage:  docker rmi [OPTIONS] IMAGE [IMAGE...]

Remove one or more images
```

**Comment by rubiin** [comment]
Let me do a fresh install and get back to you. I may need to update the readme as well

**Reply by rubiin** [reply, marked as answer]
@fac3m4n remove `NestServeStaticModule` from sharedmodule. See latest commit
